core/src/main/java/org/apache/gravitino/catalog/CatalogManager.java

```diff
@@ -466,115 +485,134 @@ public void testConnection(
-      CatalogEntity dummyEntity =
-          CatalogEntity.builder()
-              .withId(DUMMY_ID.id())
-              .withName(ident.name())
-              .withNamespace(ident.namespace())
-              .withType(type)
-              .withProvider(provider)
-              .withComment(comment)
-              .withProperties(StringIdentifier.newPropertiesWithId(DUMMY_ID, mergedConfig))
-              .withAuditInfo(
-                  AuditInfo.builder()
-                      .withCreator(creator)
-                      .withCreateTime(now)
-                      .withLastModifier(creator)
-                      .withLastModifiedTime(now)
-                      .build())
-              .build();
-
-      CatalogWrapper wrapper = createCatalogWrapper(dummyEntity, mergedConfig);
-      wrapper.doWithCatalogOps(
-          c -> {
-            c.testConnection(ident, type, provider, comment, mergedConfig);
-            return null;
-          });
-    } catch (GravitinoRuntimeException e) {
-      throw e;
-    } catch (Exception e) {
-      LOG.warn("Failed to test catalog creation {}", ident, e);
-      if (e instanceof RuntimeException) {
-        throw (RuntimeException) e;
-      }
-      throw new RuntimeException(e);
-    }
+            Map<String, String> mergedConfig = buildCatalogConf(provider, properties);
+            Instant now = Instant.now();
+            String creator = PrincipalUtils.getCurrentPrincipal().getName();
+            CatalogEntity dummyEntity =
+                CatalogEntity.builder()
+                    .withId(DUMMY_ID.id())
+                    .withName(ident.name())
+                    .withNamespace(ident.namespace())
+                    .withType(type)
+                    .withProvider(provider)
+                    .withComment(comment)
+                    .withProperties(StringIdentifier.newPropertiesWithId(DUMMY_ID, mergedConfig))
+                    .withAuditInfo(
+                        AuditInfo.builder()
+                            .withCreator(creator)
+                            .withCreateTime(now)
+                            .withLastModifier(creator)
+                            .withLastModifiedTime(now)
+                            .build())
+                    .build();
+
+            CatalogWrapper wrapper = createCatalogWrapper(dummyEntity, mergedConfig);
+            return wrapper.doWithCatalogOps(
+                c -> {
+                  c.testConnection(ident, type, provider, comment, mergedConfig);
+                  return null;
+                });
+          } catch (GravitinoRuntimeException e) {
+            throw e;
+          } catch (Exception e) {
+            LOG.warn("Failed to test catalog creation {}", ident, e);
+            if (e instanceof RuntimeException) {
+              throw (RuntimeException) e;
+            }
+            throw new RuntimeException(e);
+          }
+        });
```

Comment:
Do we need add lock for this part, I feel it is not so necessary, what do you think? @mchades

Reply:
Agree. There should be no race condition for a mocked catalog. And the test result only represents the status at that time.

Reply:
I see.